zone/fearpath.cpp

```diff
@@ -29,17 +29,52 @@ extern Zone* zone;
 
 #define FEAR_PATHING_DEBUG
 
+int Mob::GetFleeRatio(Mob* other)
+{
+	int flee_ratio = GetSpecialAbility(SpecialAbility::FleePercent); // if a special SpecialAbility::FleePercent exists
+	Mob *hate_top = GetHateTop();
+
+	if (other != nullptr) {
+		hate_top = other;
+	}
+
+	if (!hate_top) {
+		return 0;
+	}
+
+	// If no special flee_percent check for Gray or Other con rates
+	if (GetLevelCon(hate_top->GetLevel(), GetLevel()) == ConsiderColor::Gray && flee_ratio == 0 && RuleB(Combat, FleeGray) &&
```

Comment:
Since these both require flee_ratio to be 0 before triggering I think it makes sense to enclose them in a if(flee_ratio) block.

As it is right now, you're always checking the GetLevelCon stuff even if the flee ratio != 0.